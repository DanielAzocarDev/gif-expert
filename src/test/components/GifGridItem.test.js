import { shallow } from "enzyme"
import GifGridItem from '../../components/GifGridItem';


describe('Test <GifGridItem/>',() => {

  const title = 'Hola test'
  const url = 'https//:localhost.3000/algo'
  const wrapper = shallow(<GifGridItem title={title} url={url} />)

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('should have a <p> with title', () => {
    const p = wrapper.find('p')

    expect(p.text().trim()).toBe(title)
  })

  test('should have img with props url and title', () => {
    const img = wrapper.find('img')

    expect(img.prop('src')).toBe(url)
    expect(img.prop('alt')).toBe(title)
  })

  test('should have a div with class animate__fadeIn', () => {
    const div = wrapper.find('div')

    const className =div.prop('className')
    expect(className.includes('animate__fadeIn')).toBe(true)
  })
})