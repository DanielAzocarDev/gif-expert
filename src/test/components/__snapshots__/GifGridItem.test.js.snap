// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifGridItem/> should render correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn animate__delay-2s"
>
  <img
    alt="Hola test"
    src="https//:localhost.3000/algo"
  />
  <p>
    Hola test
  </p>
</div>
`;
